mgmtfn/dockplugin/netDriver.go

```diff
@@ -523,3 +630,171 @@ func GetDockerNetworkName(nwID string) (string, string, string, error) {
 
 	return tenantName, netName, serviceName, nil
 }
+
+// FindGroupFromTag finds the group that has matching tag
+func FindGroupFromTag(epgTag string) (*mastercfg.EndpointGroupState, error) {
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return nil, err
+	}
+	epgCfg := &mastercfg.EndpointGroupState{}
+	epgCfg.StateDriver = stateDriver
+	epgList, err := epgCfg.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	var epg *mastercfg.EndpointGroupState
+	found := false
+	for _, epgP := range epgList {
+		epg = epgP.(*mastercfg.EndpointGroupState)
+		if epg.GroupTag == epgTag {
+			found = true
+			break
+		}
+	}
+	if !found {
+		err = errors.New("Couldn't find group matching the tag")
+		return nil, err
+	}
+	return epg, nil
+}
+
+// FindNetworkFromTag finds the network that has matching tag
+func FindNetworkFromTag(nwTag string) (*mastercfg.CfgNetworkState, error) {
```

Comment:
This is called when docker network is created by user. It is configuration related and not frequent. We could enforce the tag to be contiv network/epg name + tenant name, and make this lookup more efficient. But it will be a dependency on the name formats and we wanted to avoid that.